release: prepare for 0.4.1 (#1465)
bitcoin-core/secp256k1: CHANGELOG.md

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

## [0.4.1] - 2023-12-21

#### Changed
 - The point multiplication algorithm used for ECDH operations (module `ecdh`) was replaced with a slightly faster one.
 - Optional handwritten x86_64 assembly for field operations was removed because modern C compilers are able to output more efficient assembly. This change results in a significant speedup of some library functions when handwritten x86_64 assembly is enabled (`--with-asm=x86_64` in GNU Autotools, `-DSECP256K1_ASM=x86_64` in CMake), which is the default on x86_64. Benchmarks with GCC 10.5.0 show a 10% speedup for `secp256k1_ecdsa_verify` and `secp256k1_schnorrsig_verify`.

#### ABI Compatibility

> The ABI is compatible with versions 0.4.0 and 0.3.x.

Author: added the word "backward"

> This release is compatible with the ABI of 0.4.0 and 0.3.x.

Reviewer: nit: Use same wording as below?  

```suggestion
The ABI is compatible with versions 0.4.0 and 0.3.x.
```
